Description: Don't try to lock and synchronise threads on kFreeBSD.
Bug-Debian: http://bugs.debian.org/705435
Origin: http://svnweb.freebsd.org/ports/head/audio/pulseaudio/files/patch-src_daemon_main.c?revision=231972&view=markup&pathrev=231972
Author: Joe Marcus Clarke <marcus@FreeBSD.org>
Author: Steven Chamberlain <steven@pyro.eu.org>
Last-Update: 2013-04-24
Applied-Upstream: 6.0, commit:69fae3e33ad024be39fb3da0643314abf4e59980

--- a/src/daemon/main.c
+++ b/src/daemon/main.c
@@ -743,6 +743,10 @@
          * first take the autospawn lock to make things
          * synchronous. */
 
+        /* This locking and thread synchronisation code doesn't work reliably
+         * on kFreeBSD (Debian bug #705435), or in upstream FreeBSD ports
+         * (bug reference: ports/128947, patched in SVN r231972). */
+#if !defined(__FreeBSD__) && !defined(__FreeBSD_kernel__)
         if ((autospawn_fd = pa_autospawn_lock_init()) < 0) {
             pa_log("Failed to initialize autospawn lock");
             goto finish;
@@ -754,6 +758,7 @@
         }
 
         autospawn_locked = true;
+#endif
     }
 
     if (conf->daemonize) {
